hbase_sql/src/main/java/com/splicemachine/derby/stream/spark/SparkDataSet.java

```diff
@@ -774,20 +780,24 @@ public static DataSet toSpliceLocatedRow(JavaRDD<Row> rdd, OperationContext cont
     }
 
     @SuppressWarnings({ "unchecked", "rawtypes" })
-    public DataSet<ExecRow> writeAvroFile(int[] baseColumnMap, int[] partitionBy, String location,  String compression,
-                                                OperationContext context) {
+    public DataSet<ExecRow> writeAvroFile(DataSetProcessor dsp,
+                                          int[] partitionBy,
+                                          String location,
+                                          String compression,
+                                          OperationContext context) {
         try {
 
-            StructType schema = ExternalTableUtils.supportAvroDateType(context.getOperation().getExecRowDefinition().schema(),"a");
+            StructType dataSchema = null;
+            StructType tableSchema = context.getOperation().getExecRowDefinition().schema();
+            dataSchema = ExternalTableUtils.getDataSchema(dsp, tableSchema, partitionBy, location, "a");
+
+            if (dataSchema == null)
+                dataSchema = tableSchema;
 
             Dataset<Row> insertDF = SpliceSpark.getSession().createDataFrame(
                     rdd.map(new SparkSpliceFunctionWrapper<>(new CountWriteFunction(context))).map(new LocatedRowToRowAvroFunction()),
-                    schema);
+                    dataSchema);
 
-            List<Column> cols = new ArrayList();
-            for (int i = 0; i < baseColumnMap.length; i++) {
-                cols.add(new Column(ValueRow.getNamedColumn(baseColumnMap[i])));
-            }
             List<String> partitionByCols = new ArrayList();
             for (int i = 0; i < partitionBy.length; i++) {
                 partitionByCols.add(ValueRow.getNamedColumn(partitionBy[i]));
```

Comment:
I think we need to handle partitioned columns in Orc and Avro.

Reply:
@changli6 Yes. The removed lines are useless.

Reply:
The name was not modified for the partitioned columns?  We may treat them same as in write ParquetFile : partitionByCols.add(dataSchema.fields()[partitionBy[i]].name());